### ISSUE TITLE
Interrupt soft-masking could keep MSR[EE] enabled if an external interrupt hits

### ISSUE THREAD

**Issue by npiggin** [linuxppc/issues#380]
Balbir had a patch to make soft-NMI watchdog a bit more capable by acknowledging the interrupt and marking it pending without disabling MSR[EE]

https://patchwork.ozlabs.org/project/linuxppc-dev/patch/20170105045538.32658-3-bsingharora@gmail.com/

It's not clear that's really worth it. The SMP checking covers most of the soft-NMI watchdog's failings. Maybe it would help profiling in high-interrupt-rate workloads? A soft-NMI IPI could be a little more interesting if that was implemented, but between system reset IPI and r1 stack saving, it may not be a big improvement.

Opening this to keep track of the idea for now so the patchwork series can be closed out.

**Comment by npiggin** [comment]
Now I have your attention, a question - does it make much sense to keep such things around (either here or in patchwork) as just-in-case ideas? If they are not so obviously good to implement, maybe they will just never go away and keep cluttering up our trackers. There seem to be a few such things in patchwork that we could close if we either move them somewhere else or just decide to close them.

**Comment by npiggin** [comment]
Yeah it may have other issues, but certainly the asm no longer applies so I marked it N/A. I thought you might be keeping it around in patchwork to keep the idea alive.

**Comment by mpe** [comment]
Yeah I kind of was keeping it in patchwork so I could come back to it one day. But using an issue like this is better than leaving it in patchwork for ever.
